string in_string
---
bool ok
string out_string
